### PR TITLE
adds `templates publish` command

### PR DESCRIPTION
**Pull request by samruddhikhandale**
Based on top of [samruddhikhandale/templates-package](https://github.com/devcontainers/cli/pull/193) branch.

Adds a `template publish` CLI command according to the [spec](https://github.com/devcontainers/spec/blob/main/proposals/devcontainer-templates.md)

image

Sample - publishing templates

image

image

image

### CODE REVIEW

File: src/spec-node/templatesCLI/publish.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    const cliHost = await getCLIHost(cwd, loadNativeModule);
+    const output = createLog({
+        logLevel: mapLogLevel(inputLogLevel),
+        logFormat: 'text',
+        log: (str) => process.stdout.write(str),
+        terminalDimensions: undefined,
+    }, pkg, new Date(), disposables);
+
+    // Package templates
+    const outputDir = path.join(os.tmpdir(), '/templates-output');
```
Comment (chrmarti) [inline src/spec-node/templatesCLI/publish.ts:53]: Make this a unique folder name. Appending `Date.now()` might be enough, but I have started using the `uuid` package in vscode-remote-containers to be on the safe side, e.g.: `/templates-output-${uuid.v4()}`.

Could you apply the same fix in the features code?

**Review by chrmarti — CHANGES_REQUESTED** [review]
LGTM! Left one comment to avoid tmp folder conflicts.

File: src/spec-node/templatesCLI/publish.ts (hunk 1 above)
Comment (samruddhikhandale) [reply src/spec-node/templatesCLI/publish.ts]: Sure, added changes. 👍

**Review by chrmarti — APPROVED** [review]
LGTM!

**Pull request merged**